src/main/java/org/littleshoot/proxy/impl/ProxyUtils.java

```diff
@@ -637,4 +637,30 @@ public static FullHttpResponse createFullHttpResponse(HttpVersion httpVersion,
 
         return response;
     }
+
+    /**
+     * Given an HttpHeaders instance, removes 'sdch' from the 'Accept-Encoding'
+     * header list (if it exists) and returns the modified instance.
+     *
+     * Removes all occurrences of 'sdch' from the 'Accept-Encoding' header.
+     * @param headers The headers to modify.
+     */
+    public static HttpHeaders removeSdchEncoding(HttpHeaders headers) {
+        if (headers == null) {
+            throw new IllegalArgumentException("Headers is null");
+        }
+
+        String ae = headers.get(HttpHeaders.Names.ACCEPT_ENCODING);
+        if (StringUtils.isNotBlank(ae)) {
+            // The former regex should remove occurrences of 'sdch' while the
+            // latter regex should take care of the dangling comma case when
+            // 'sdch' was the first element in the list and there are other
+            // encodings.
+            String noSdch = ae.replaceAll(",? *(sdch|SDCH)", "").replaceFirst("^ *, *", "");
+            headers.set(HttpHeaders.Names.ACCEPT_ENCODING, noSdch);
```

Comment:
That's a great point; I didn't realize this issue.

The short answer to your question is that both scenarios are possible.  Here's an excerpt from my analysis:

```java
HttpHeaders headers = new DefaultHttpHeaders();

headers.add(HttpHeaders.Names.ACCEPT_ENCODING, "gzip");
headers.add(HttpHeaders.Names.ACCEPT_ENCODING, "deflate");
headers.add(HttpHeaders.Names.ACCEPT_ENCODING, "br");
headers.add(HttpHeaders.Names.ACCEPT_ENCODING, "foo, bar, baz");

// Prints "gzip" since it's the first value for the header.  Behavior follows from here: https://github.com/netty/netty/blob/netty-4.0.42.Final/codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java#L268-L289
System.out.println(headers.get(HttpHeaders.Names.ACCEPT_ENCODING));

// Prints "gzip", "deflate", "br" and "foo, bar, baz" each on its own line.  Behavior follows from here: https://github.com/netty/netty/blob/netty-4.0.42.Final/codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java#L297-L314
for (String encoding : headers.getAll(HttpHeaders.Names.ACCEPT_ENCODING)) {
  System.out.println(encoding);
}
```

Given that DefaultHttpHeaders.get() returns only the first header, I think this method should generally be avoided in favor of DefaultHttpHeaders.getAll() unless one knows for sure that there is, at most, one value mapped to the header.

For my specific case, I'm primarily concerned with how the headers are structured when I go through BrowserMob Proxy.  From what I was able to deduce, it appears that I'm being provided a single value to ACCEPT_ENCODING mapped to "gzip, deflate, sdch, br" as opposed to it being broken out individually for each encoding.  Unfortunately, that means I can't use the HttpHeaders.contains() method to check for the presence of the "sdch" encoding.

With all that said, if we want to properly support all the ways the encoding header can be defined either as...

1. Multiple distinct singular encodings (e.g. "gzip", "deflate", "br")
2. A single comma delineated string of encodings (e.g. "gzip, deflate, br")
3. Any combination of 1 and 2 (e.g. "gzip", "deflate, br")

...I propose the following:

1. Iterate through all the encodings that map to ACCEPT_ENCODING and construct a single comma delineated string of encodings.
2. Perform the current regex replacement that we have to remove any and all occurrences of "sdch".
3. Set the value of the ACCEPT_ENCODING header to the string that resulted after step 2.

What are your thoughts on this?